app/components/glimmer/route-graphic.js

```diff
@@ -3,59 +3,67 @@ import { cached } from '@glimmer/tracking';
 import Graphic from '@arcgis/core/Graphic';
 import Polyline from '@arcgis/core/geometry/Polyline';
 import SimpleLineSymbol from '@arcgis/core/symbols/SimpleLineSymbol';
-import { registerDestructor } from '@ember/destroyable';
 
 export default class RouteGraphicComponent extends Component {
+  createGraphic = () => {
+    console.log('updating graphic');
 
-  constructor() {
-    super(...arguments);
-
-    registerDestructor(this, this.destroyGraphic);
-  }
-
-  @cached
-  get graphic() {
     if (!this.args.layer) {
       return null;
     }
 
-    // console.log('Setting up graphic: ', this.args.routeData);
-    const polyline = new Polyline({
-      paths: this.args.routeData.paths,
-    });
+    (async () => {
+      await Promise.resolve();
+
+      // console.log('Setting up graphic: ', this.args.routeData);
+      const polyline = new Polyline({
+        paths: this.args.routeData.paths,
+      });
 
-    const lineSymbol = new SimpleLineSymbol({
-      color: this.args.routeData.color,
-      width: 4,
-    });
+      const lineSymbol = new SimpleLineSymbol({
+        color: this.args.routeData.color,
+        width: 4,
+      });
 
-    const graphic = new Graphic({
-      geometry: polyline,
-      symbol: lineSymbol,
-    });
+      const graphic = new Graphic({
+        geometry: polyline,
+        symbol: lineSymbol,
+      });
 
-    //console.log(`Adding graphic ${this.args.routeData.id}`);
-    this.args.layer.add(graphic);
+      // WARNING: this doesn't remove if any tracked data in this getter changes
+      //          a resource would be better for property-based destruction.
+      //
+      //          Or a separate component for the "graphic"
+      //console.log(`Adding graphic ${this.args.routeData.id}`);
+      this.args.layer.add(graphic);
 
-    return graphic;
+      this.graphic = graphic;
+    })();
   };
 
-  updateGraphic = () => {
-    if (!this.graphic) {
-      return;
-    }
-    //console.log('updating graphic ' + this.args.routeData.id);
+  changeColor = () => {
+    // Always invalidate when color changes
+    this.args.routeData.color;
+
+    (async () => {
+      await Promise.resolve();
```

Comment:
?

Reply:
since auto-tracking is synchronous, this detaches from auto-tracking